diff --git a/C:/Users/yuntao/AppData/Local/Temp/cpython-3.12.3/Modules/socketmodule.h b/external/io/include/socketmodule.h
index f5ca004..9005583 100644
--- a/C:/Users/yuntao/AppData/Local/Temp/cpython-3.12.3/Modules/socketmodule.h
+++ b/external/io/include/socketmodule.h
@@ -1,5 +1,8 @@
+#pragma once
 /* Socket module header file */
 
+// Modified by CCP ehf.
+
 /* Includes needed for the sockaddr_* symbols below */
 #ifndef MS_WINDOWS
 #ifdef __VMS
@@ -98,6 +101,8 @@ typedef int socklen_t;
 #  include <asm/types.h>
 # endif
 # include <linux/netlink.h>
+#elif defined(HAVE_NETLINK_NETLINK_H)
+# include <netlink/netlink.h>
 #else
 #  undef AF_NETLINK
 #endif
@@ -224,7 +229,7 @@ extern "C" {
 #endif
 
 /* Python module and C API name */
-#define PySocket_MODULE_NAME    "_socket"
+#define PySocket_MODULE_NAME    "_carbonsocket"
 #define PySocket_CAPI_NAME      "CAPI"
 #define PySocket_CAPSULE_NAME   PySocket_MODULE_NAME "." PySocket_CAPI_NAME
 
@@ -311,7 +316,9 @@ typedef union sock_addr {
    like the address family, which is used to decode socket address
    arguments properly. */
 
-typedef struct {
+typedef struct uv_handle_s uv_handle_t; // forward declare
+typedef struct uv_loop_s uv_loop_t; // forward declare
+typedef struct PySocketSockObject_t {
     PyObject_HEAD
     SOCKET_T sock_fd;           /* Socket file descriptor */
     int sock_family;            /* Address family, e.g., AF_INET */
@@ -320,9 +327,10 @@ typedef struct {
     PyObject *(*errorhandler)(void); /* Error handler; checks
                                         errno, returns NULL and
                                         sets a Python exception */
-    _PyTime_t sock_timeout;     /* Operation timeout in seconds;
+    PyTime_t sock_timeout;     /* Operation timeout in seconds;
                                         0.0 means non-blocking */
     struct _socket_state *state;
+	uv_handle_t* uv_handle;     /* the libuv handle associated with this socket object */
 } PySocketSockObject;
 
 /* --- C API ----------------------------------------------------*/
@@ -377,12 +385,81 @@ typedef struct {
 
 /* C API for usage by other Python modules.
  * Always add new things to the end for binary compatibility. */
+
+/**
+*   A callback method that gets called with data from received packets before they get received on the Python side.
+*   If one of these returns a truthy value, the packet gets "swallowed", and will not be received through the Python
+*   `recvpacketoob` call.
+*/
+typedef int(*OobDataCallback)(long long descriptor, const char* data, int len, const char* OOBdata, int OOBLen );
+
 typedef struct {
     PyTypeObject *Sock_Type;
     PyObject *error;
     PyObject *timeout_error;
+
+    /**
+    *   @brief Unblock any IO blocked tasklets that have finished their blocking operations.
+    */
+    void (*dispatch)();
+
+    /**
+    *  @brief Add a callback for handling packets when they are received. The callbacks will get called in order before the packet gets returned to Python.
+    *  If one of the callbacks returns `true`, no further callbacks will be called and the packet will not make its way to the receiver,
+    *  which will continue to wait for the next packet.
+    *  @param OobDataCallback the callback to register
+    */
+
+    void (*add_oob_data_callback)(OobDataCallback);
+    /**
+    *  @brief Remove a callback that was added using `add_oob_data_callback`. The callback will no longer get called to process received packets.
+    *  @param OobDataCallback the callback to remove
+    */
+    void (*remove_oob_data_callback)(OobDataCallback);
+
+    /**
+    *  @brief Formats a macho packet containing the data as well as any out-of-band data.
+    *  The results are placed inside the buffer pointed to by the `buf` parameter.
+    *  The buffer is assumed to be large enough to hold the formatted packet.
+    *  @param buf The buffer into which to place the formatted packet
+    *  @param data The data to place into the packet
+    *  @param data The length of the data to place into the packet in bytes.
+    *  @param OOBData out-of-band data to place into the packet, or nullptr if there is no OOBData.
+    *  @param OOBDataLen Length of the out-of-band data in bytes.
+    *  @returns The size of the formatted packet in bytes
+    */
+    int (*format_packet)(char* buf, const char* data, const uint32_t dataLen, const char* OOBData, const uint32_t OOBLen);
+
+    /**
+    *  @brief Sends a macho packet. Assumes the packet is already formatted according to FormatPacket.
+    *  @param fd The file descriptor of the socket to use.
+    *  @param data A formatted macho packet
+    *  @param len The length of the packet to send in bytes
+    */
+    int (*send_formatted_packet)(const long long fd, const char* data, const unsigned int len);
+
+    /**
+    *  @brief Creates a formatted macho packet, and sends it.
+    *  @param fd The file descriptor of the socket to use.
+    *  @param data A formatted macho packet
+    *  @param len The length of the data to send in bytes
+    *  @param OOBData out-of-band data to place into the packet, or nullptr if there is no OOBData.
+    *  @param OOBDataLen Length of the out-of-band data in bytes.
+    */
+    int (*send_packet)(const long long fd, const char* data, const unsigned int len, const char* OOBData, const unsigned int OOBLen);
+
+    /**
+    *  @returns a pointer to the UV loop used for managing the socket IO events.
+    *  @note Allows other libraries to share the UV loop for potentially improved performance.
+    */
+	uv_loop_t* (*get_uv_loop)();
+
 } PySocketModule_APIObject;
 
+
+/**
+*  Import the module and return a pointer to the C [capsule](https://docs.python.org/3.12/c-api/capsule.html) object.
+*/
 #define PySocketModule_ImportModuleAndAPI() PyCapsule_Import(PySocket_CAPSULE_NAME, 1)
 
 #ifdef __cplusplus
